node/flatpak_node_generator/providers/npm.py

```diff
@@ -459,56 +497,57 @@ def _finalize(self) -> None:
-                        f' > {target}.new'
-                    )
-                    patch_commands[lockfile].append(f'mv {target}{{.new,}}')
-
-        patch_all_commands: List[str] = []
-        for lockfile in self.all_lockfiles:
-            patch_dest = (
-                self.gen.data_root / 'patch' / self.relative_lockfile_dir(lockfile)
-            )
-            # Don't use with_extension to avoid problems if the package has a . in its name.
-            patch_dest = patch_dest.with_name(patch_dest.name + '.sh')
+                with open(lockfile) as fp:
+                    lockfile_v1 = json.load(fp)['lockfileVersion'] == 1
+
+                if lockfile_v1:
+                    for path, name_source in sources.items():
+                        GIT_URL_PREFIX = 'git+'
+                        name, source = name_source
+                        new_version = f'{path}#{source.commit}'
+                        data['package.json'][name] = new_version
+                        data['package-lock.json'][source.original] = new_version
+
+                        if source.original.startswith(GIT_URL_PREFIX):
+                            data['package-lock.json'][
+                                source.original[len(GIT_URL_PREFIX) :]
+                            ] = new_version
+
+                    for filename, script in scripts.items():
+                        target = Path('$FLATPAK_BUILDER_BUILDDIR') / prefix / filename
+                        script = (
+                            textwrap.dedent(script.lstrip('\n'))
+                            .strip()
+                            .replace('\n', '')
+                        )
+                        json_data = json.dumps(data[filename])
+                        patch_commands[lockfile].append(
+                            'jq'
+                            ' --arg buildroot "$FLATPAK_BUILDER_BUILDDIR"'
+                            f' --argjson data {shlex.quote(json_data)}'
+                            f' {shlex.quote(script)} {target}'
+                            f' > {target}.new'
+                        )
+                        patch_commands[lockfile].append(f'mv {target}{{.new,}}')
+
+        if len(patch_commands) > 0:
+            patch_all_commands: List[str] = []
+            for lockfile in self.all_lockfiles:
+                patch_dest = (
+                    self.gen.data_root / 'patch' / self.relative_lockfile_dir(lockfile)
+                )
+                # Don't use with_extension to avoid problems if the package has a . in its name.
+                patch_dest = patch_dest.with_name(patch_dest.name + '.sh')
 
-            self.gen.add_script_source(patch_commands[lockfile], patch_dest)
-            patch_all_commands.append(f'"$FLATPAK_BUILDER_BUILDDIR"/{patch_dest}')
+                self.gen.add_script_source(patch_commands[lockfile], patch_dest)
+                patch_all_commands.append(f'$FLATPAK_BUILDER_BUILDDIR/{patch_dest}')
 
```

Comment:
why was the quote removed? That came from a PR to solve a bug when the pwd has spaces in it, same for below

Reply:
I didn't notice that change when I rebased, added both back now.